### PR TITLE
xeus-cling: fix improper linking with LLVM

### PR DESCRIPTION
**Pull request by thomasjm**
This is a fix for #306782. The solution, as recommended by the upstream Cling maintainer, is to turn off the shared library flag for LLVM when building Cling.

The second commit here adds a smoke check for `xeus-cling`, using Papermill to run a test notebook.

I also changed the default `xeus-cling` derivation to include the various library flags it expects, since it's fairly useless without them. You can still get the unwrapped `xeus-cling` version via the passthru `xeus-cling.unwrapped`.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by wegank — APPROVED** [review]

File: pkgs/applications/editors/jupyter-kernels/xeus-cling/default.nix [hunk 1]
```diff
@@ -11,48 +14,81 @@
...
+    displayName = builtins.replaceStrings ["c++"] ["C++ "] std;
+    argv = [
+      "${xeus-cling}/bin/xcpp"
+      "-std=${std}"
+      "-f" "{connection_file}"
+    ];
+    language = "cpp";
+    logo32 = fetchurl {
+      url = "https://upload.wikimedia.org/wikipedia/commons/thumb/1/18/ISO_C%2B%2B_Logo.svg/32px-ISO_C%2B%2B_Logo.svg.png";
+      hash = "sha256-cr0TB8/j2mkcFhfCkz9F7ZANOuTlWA2OcWtDcXyOjHw=";
```
Comment (qbisi) [inline pkgs/applications/editors/jupyter-kernels/xeus-cling/default.nix:77]: new hash should be "sha256-+TKtwXybKw4oAHfgOsDxvL4ucItPguF76HJHdFTd3s0=" (maybe file changed)

File: pkgs/applications/editors/jupyter-kernels/xeus-cling/default.nix [hunk 2]
```diff
@@ -11,48 +14,81 @@
...
+    };
+    logo64 = fetchurl {
+      url = "https://upload.wikimedia.org/wikipedia/commons/thumb/1/18/ISO_C%2B%2B_Logo.svg/64px-ISO_C%2B%2B_Logo.svg.png";
+      hash = "sha256-nZtJ4bR7GmQttvqEJC9KejOxphrjjxT36L9yOIITFLk=";
```
Comment (qbisi) [inline pkgs/applications/editors/jupyter-kernels/xeus-cling/default.nix:81]: should be "sha256-7SjOcSaSPUHIKnjBxMdn+KSjviL69IXhX7eJsacYeGE="

Comment (thomasjm) [reply pkgs/applications/editors/jupyter-kernels/xeus-cling/default.nix]: Done

File: pkgs/applications/editors/jupyter-kernels/xeus-cling/default.nix (hunk 1 above)
Comment (thomasjm) [reply pkgs/applications/editors/jupyter-kernels/xeus-cling/default.nix]: Done

**Pull request merged**